podman/config.py

```diff
@@ -0,0 +1,80 @@
+"""Read containers.conf file."""
+import os
+import pathlib
+import urllib.parse
+from pathlib import Path
+from typing import Dict, Optional
+
+try:
+    import toml
+except ImportError:
+    import pytoml as toml
+
+
+class ServiceConnection:
+    """ServiceConnection defines a connection to the Podman service."""
+
+    def __init__(self, name: str, attrs: Dict[str, str]):
+        """Create a Podman ServiceConnection."""
+        self.name = name
+        self.attrs = attrs
+
+    def __repr__(self) -> str:
+        return f"""<{self.__class__.__name__}: '{self.id}'>"""
+
+    @property
+    def id(self) -> str:  # pylint: disable=invalid-name
+        """Returns id for service connection."""
+        return self.name
+
+    @property
+    def url(self) -> urllib.parse.ParseResult:
+        """Returns URL for service connection."""
+        return urllib.parse.urlparse(self.attrs.get("uri"))
+
+    @property
+    def identity(self) -> Path:
+        """Returns Path to identity file for service connection."""
+        return pathlib.Path(self.attrs.get("identity"))
+
+
+class PodmanConfig:
+    """PodmanConfig provides a representation of the containers.conf file."""
+
+    def __init__(self, path: str = None):
+        """Read Podman configuration from users XDG_CONFIG_HOME."""
+        # Python 3.6 does not include cached_property therefore must work is done in __init__.
+
+        if path is None:
+            path = os.environ.get("XDG_CONFIG_HOME", f"{Path.home()}/.config/containers")
```

Comment:
I think this might be wrong.  Did you intend something like this:

```python
path = Path(os.environ.get("XDG_CONFIG_HOME", f"{Path.home()}/.config"))
path = path.joinpath("containers")
```

Reply:
Good catch, next push has no platform dependencies:

```
            home = Path(os.environ.get("XDG_CONFIG_HOME", Path.home().name))
            self.path = home / ".config" / "containers" / "containers.conf"
```